tasks.py

```diff
@@ -0,0 +1,115 @@
+"""Module of Invoke tasks regarding CODE QUALITY to be invoked from the command line. Try
+invoke --list
+from the command line for a list of all available commands.
+"""
+import os
+
+from invoke import task
+
+POSIX = os.name == "posix"
+
+
+@task
+def black(command, checkonly=False):
+    """Runs black (autoformatter) on all .py files recursively"""
+    print(
+        """
+Running Black the Python code formatter
+=======================================
+"""
+    )
+    cmd = "black --check --diff ." if checkonly else "black ."
+    command.run(cmd, echo=True, pty=POSIX)
+
+
+@task
+def isort(command, checkonly=False):
+    """Runs isort (import sorter) on all .py files recursively"""
+    print(
+        """
+Running isort the Python code import sorter
+===========================================
+"""
+    )
+    cmd = "isort --check-only --diff ." if checkonly else "isort ."
+    command.run(cmd, echo=True, pty=POSIX)
+
+
+@task
+def lint(
+    command,
+):
+    """Runs flake8 plugin flakeheaven (linter) on all .py files recursively"""
+    print(
+        """
+Running flakeheaven, a Python code linter
+===================================
+"""
+    )
+    command.run("flakeheaven lint", echo=True, pty=POSIX)
+
+
+@task(pre=[black, isort, lint])
+def style(
+    command,
+):
+    """Runs black, isort, and flake8
+    Arguments:
+        command {[type]} -- [description]
+    """
+    # If we get to this point all tests listed in 'pre' have  passed
+    # unless we have run the task with the --warn flag
+    if not command.config.run.warn:
+        print(
+            """
+All Style Checks Passed Successfully
+====================================
+"""
+        )
+
+
+@task
+def pytest(command, options):
+    """Runs pytest to identify failing tests and doctests"""
+
+    print(
+        """
+Running pytest the test framework
+=================================
+"""
+    )
+    command.run(f"python -m pytest {options} .", echo=True, pty=POSIX)
+
+
+@task
+def docs(
+    command,
+):
+    """Runs Sphinx to build the docs locally for testing"""
+
+    print(
+        """
+Running Sphinx to test the docs building
+========================================
+"""
+    )
+    command.run("sphinx-build -b html docs docs/_build/html", echo=True, pty=POSIX)
```

Comment:
We should use build scripts instead or the build directories won't be deleted and cause a bunch of recursive build stuff. We can change this to run the build_docs.sh script instead, but we will need to add a different solution for windows.